From efa5b0a190f9d71e4d4b9447e078a30c2a8849fe Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@evolve-os.com>
Date: Sat, 27 Sep 2014 20:20:22 +0100
Subject: [PATCH] include: Include stdbool header before using 'bool'

Signed-off-by: Ikey Doherty <ikey@evolve-os.com>
---
 include/tdb.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/tdb.h b/include/tdb.h
index 5ea5e60..03e429c 100644
--- a/include/tdb.h
+++ b/include/tdb.h
@@ -31,6 +31,7 @@ extern "C" {
 #endif
 
 #include <signal.h>
+#include <stdbool.h>
 
 /**
  * @defgroup tdb The tdb API
-- 
1.8.4

