From 2ebc63c7de89d2249d6f26baf1fc7f3069a479a8 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solusos.com>
Date: Sat, 10 Aug 2013 15:34:49 +0100
Subject: [PATCH] Apply SolusOS default branding

---
 src/panel_plugin.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/panel_plugin.cpp b/src/panel_plugin.cpp
index 91bed49..683a0f9 100644
--- a/src/panel_plugin.cpp
+++ b/src/panel_plugin.cpp
@@ -46,8 +46,8 @@ PanelPlugin::PanelPlugin(XfcePanelPlugin* plugin) :
 	m_plugin(plugin),
 	m_menu(NULL),
 	m_button_title(get_button_title_default()),
-	m_button_icon_name("xfce4-whiskermenu"),
-	m_button_title_visible(false),
+	m_button_icon_name("/usr/share/pixmaps/SolusOS.png"),
+	m_button_title_visible(true),
 	m_button_icon_visible(true)
 {
 	// Load settings
-- 
1.7.10.4

