From 97650f55f212266fcb5b303971315c5b819b3264 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solusos.com>
Date: Mon, 2 Sep 2013 17:19:40 +0100
Subject: [PATCH] Disable test program (segfaults during build)

---
 src/Makefile.am |   12 ------------
 1 file changed, 12 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 7c6e21b..1e11398 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -59,18 +59,6 @@ AM_CPPFLAGS =						\
 	-DDATADIR="\"$(uidir)\""			\
 	$(NULL)
 
-noinst_PROGRAMS = test-timezone
-
-test_timezone_SOURCES = test-timezone.c tz.c tz.h
-test_timezone_LDADD = $(LIBTIMEZONEMAP_LIBS) -lm
-test_timezone_CFLAGS = $(LIBTIMEZONEMAP_CFLAGS)
-
-all-local: check-local
-
-# FIXME remove "|| :" when we have all the necessary pixmaps
-check-local: test-timezone
-	$(builddir)/test-timezone $(srcdir)/data || :
-
 lib_LTLIBRARIES = libtimezonemap.la
 
 libtimezonemap_la_SOURCES =	\
-- 
1.7.10.4

