From 43e67931a4949ed60c8c1e6c834338073eb3e612 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Sat, 10 May 2014 16:17:40 +0100
Subject: [PATCH] Only remove unneeded and debug symbols from kernel modules

---
 pisi/util.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/util.py b/pisi/util.py
index 6777385..d4296a8 100644
--- a/pisi/util.py
+++ b/pisi/util.py
@@ -592,7 +592,7 @@ def strip_file(filepath, fileinfo, outpath):
         if ctx.config.values.build.generatedebug:
             ensure_dirs(os.path.dirname(outpath))
             save_elf_debug(filepath, outpath)
-        run_strip(filepath)
+        run_strip(filepath, "-g --strip-unneeded")
         return True
 
     elif "SB shared object" in fileinfo:
-- 
1.9.1

