From d547b9d9c4c943fc5edb428686c5d4e541a58a8f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Fri, 6 Jun 2014 19:31:50 +0100
Subject: [PATCH 4/5] Fix issue OS-4 in Evolve OS (breaks binary updates)

---
 pisi/relation.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/relation.py b/pisi/relation.py
index b5115a8..261b9e0 100644
--- a/pisi/relation.py
+++ b/pisi/relation.py
@@ -57,7 +57,7 @@ class Relation:
 def installed_package_satisfies(relation):
     installdb = pisi.db.installdb.InstallDB()
     pkg_name = relation.package
-    if relation.type == "pkgconfig":
+    if hasattr(relation, "type") and relation.type == "pkgconfig":
         pkg = installdb.get_package_by_pkgconfig(pkg_name)
         if pkg:
             return relation.satisfies_relation(pkg.version, pkg.release)
-- 
1.9.1

