From cc6f634d80caa443ad3502e9bee6c2b1abeb2515 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@evolve-os.com>
Date: Sun, 8 Mar 2015 02:45:23 +0000
Subject: [PATCH 2/2] That firmly bit us on the ass. Let autocollapse do this
 for us

---
 packageit.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/packageit.py b/packageit.py
index ee984ea..372c932 100644
--- a/packageit.py
+++ b/packageit.py
@@ -112,9 +112,6 @@ def packageit(ymlFile, installDIR, outputXML):
     patterns["/usr/share/icons"] = name
     patterns["/usr/share/pixmaps"] = name
     patterns["/usr/share/man"] = name
-    patterns["/usr/lib64/%s" % name] = name
-    patterns["/usr/lib32/%s" % name] = "-32bit"
-    patterns["/usr/share/%s" % name] = name
     patterns["/usr/bin"] = name
     patterns["/usr/sbin"] = name
 
-- 
1.8.4

