diff -ruNb a/src/vesa.c b/src/vesa.c
--- a/src/vesa.c	2012-07-18 10:59:39.000000000 +0100
+++ b/src/vesa.c	2013-03-28 12:12:42.200777275 +0000
@@ -50,7 +50,6 @@
 #include "mipointer.h"
 
 /* All drivers implementing backing store need this */
-#include "mibstore.h"
 
 /* Colormap handling */
 #include "micmap.h"
@@ -1081,7 +1080,6 @@
     VESADGAInit(pScrn, pScreen);
 
     xf86SetBlackWhitePixels(pScreen);
-    miInitializeBackingStore(pScreen);
     xf86SetBackingStore(pScreen);
 
     /* software cursor */
@@ -1626,7 +1624,7 @@
 {
     VESAPtr pVesa;
 
-    if (MODE_QUERY < 0 || function > MODE_RESTORE)
+    if (function < MODE_QUERY || function > MODE_RESTORE)
 	return (FALSE);
 
     pVesa = VESAGetRec(pScrn);
