diff -ruNb a/gdk/wayland/gdkwaylandwindow.h b/gdk/wayland/gdkwaylandwindow.h
--- a/gdk/wayland/gdkwaylandwindow.h	2013-06-19 22:42:45.000000000 +0100
+++ b/gdk/wayland/gdkwaylandwindow.h	2013-06-27 21:55:49.571719587 +0100
@@ -49,7 +49,7 @@
 struct wl_surface       *gdk_wayland_window_get_wl_surface       (GdkWindow *window);
 GDK_AVAILABLE_IN_ALL
 struct wl_shell_surface *gdk_wayland_window_get_wl_shell_surface (GdkWindow *window);
-
+GDK_AVAILABLE_IN_ALL
 void                     gdk_wayland_window_set_use_custom_surface (GdkWindow *window);
 
 G_END_DECLS
