From 89fb5f441d0805e001bced45c4390da59afd2bae Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Sun, 6 Jul 2014 15:58:23 +0100
Subject: [PATCH 02/11] interface: Be a fully fledged window, no close button
 under Budgie

---
 os_installer/interface.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/os_installer/interface.py b/os_installer/interface.py
index 9ccb62d..84679ec 100644
--- a/os_installer/interface.py
+++ b/os_installer/interface.py
@@ -64,7 +64,6 @@ class InstallerWindow(Gtk.Window):
         self.layout.add(self.stack)
 
         self.set_resizable(False)
-        self.set_type_hint(Gdk.WindowTypeHint.DIALOG)
         self.show_all()
 
     def _init_theme(self):
-- 
1.8.4

