From 6af3b4c49cddc9e709b02b87a97d02d962e02a3c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Wed, 9 Jul 2014 02:12:28 +0100
Subject: [PATCH 3/7] users: Ensure minimum length of 6, not 7 (OS-26)

---
 os_installer/pages/users.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/os_installer/pages/users.py b/os_installer/pages/users.py
index 1e4d300..6c86ea4 100644
--- a/os_installer/pages/users.py
+++ b/os_installer/pages/users.py
@@ -95,14 +95,14 @@ class NewUserPage(Gtk.Grid):
             pass1 = self.pword_field.get_text()
             pass2 = self.pword_field2.get_text()
 
-            if len(pass1) > PASSWORD_LENGTH:
+            if len(pass1) >= PASSWORD_LENGTH:
                 self.pword_field.set_icon_from_icon_name(Gtk.EntryIconPosition.SECONDARY, "emblem-ok-symbolic")
                 self.update_score(self.pword_field, True)
             else:
                 self.pword_field.set_icon_from_icon_name(Gtk.EntryIconPosition.SECONDARY, None)
                 self.update_score(self.pword_field, False)
                 
-            if len(pass1) > PASSWORD_LENGTH and pass1 == pass2:
+            if len(pass1) >= PASSWORD_LENGTH and pass1 == pass2:
                 self.pword_field2.set_icon_from_icon_name(Gtk.EntryIconPosition.SECONDARY, "emblem-ok-symbolic")
                 self.update_score(self.pword_field2, True)
             else:
-- 
1.9.1

