From 233c0d561388721cab26b8bcd208327f4bb28131 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Wed, 9 Jul 2014 02:14:11 +0100
Subject: [PATCH 4/5] users: Ensure only lower case user names (OS-26)

---
 os_installer/pages/users.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer/pages/users.py b/os_installer/pages/users.py
index 6c86ea4..2c514f3 100644
--- a/os_installer/pages/users.py
+++ b/os_installer/pages/users.py
@@ -130,7 +130,7 @@ class NewUserPage(Gtk.Grid):
         self.scores = dict()
         self.needed_score = 4
 
-        self.username_regex = re.compile(UNAME_REGEX, re.IGNORECASE)
+        self.username_regex = re.compile(UNAME_REGEX)
 
         row = 0
         uname_label = Gtk.Label(_("Username:"))
-- 
1.9.1

