From c72eef1fa3747f5de9ca3c6d9d945c2f57daa8ce Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Sun, 13 Jul 2014 19:54:26 +0100
Subject: [PATCH 07/10] Ensure we say alpha2 in the version string

---
 dist/install.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dist/install.conf b/dist/install.conf
index a45f01f..f65c765 100644
--- a/dist/install.conf
+++ b/dist/install.conf
@@ -1,7 +1,7 @@
 # Defines the distribution name and version..
 [Branding]
 Name=Evolve OS
-Version=alpha1
+Version=alpha2
 Theme=DarkTheme
 
 [Configuration]
-- 
1.8.4

