From 600a3f85f6f8c1db0864d5f7111a4c39fe0c1fed Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Sun, 3 Aug 2014 14:33:47 +0100
Subject: [PATCH 09/11] installer: Ensure we sync pre-umount

---
 os_installer/installer.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/os_installer/installer.py b/os_installer/installer.py
index 40c9379..be93af8 100644
--- a/os_installer/installer.py
+++ b/os_installer/installer.py
@@ -357,6 +357,7 @@ EndSection\n""" % (setup.keyboard_model, setup.keyboard_layout))
             # now unmount it
             print " --> Unmounting partitions"
             try:
+                os.system("sync")
                 os.system("umount --force /target/dev/shm")
                 os.system("umount --force /target/dev/pts")
                 os.system("umount --force /target/dev/")
-- 
1.8.4

