From 454faeca7e005b2068a1c567a0aaae288a75aab3 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Sun, 15 Jun 2014 16:56:31 +0100
Subject: [PATCH] configure: Drop 1-point version for upower requirement

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 5c7c220..3a6d53a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -31,7 +31,7 @@ PKG_CHECK_MODULES([MUTTER], [libmutter >= 3.10.1])
 PKG_CHECK_MODULES([WNCK3], [libwnck-3.0 >= 3.4.7])
 
 # Required for power-applet
-PKG_CHECK_MODULES([UPOWER], [upower-glib >= 0.9.21])
+PKG_CHECK_MODULES([UPOWER], [upower-glib >= 0.9.20])
 
 # Required for menu in the panel
 PKG_CHECK_MODULES([GMENU], [libgnome-menu-3.0 >= 3.10.1])
-- 
1.9.1

