From 62380f01f6f111ad0517076705a539bdc415522b Mon Sep 17 00:00:00 2001
From: Elias Aebi <user142@hotmail.com>
Date: Wed, 25 Jun 2014 18:54:49 +0200
Subject: [PATCH 4/5] set the icon geometry in order to fix the minimise
 animation

---
Fixes #36
Signed-off-by: Ikey Doherty <ikey.doherty@gmail.com>
---
 panel/icon-tasklist.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/panel/icon-tasklist.c b/panel/icon-tasklist.c
index f131a57..47c16fe 100644
--- a/panel/icon-tasklist.c
+++ b/panel/icon-tasklist.c
@@ -41,6 +41,9 @@ static gboolean button_draw(GtkWidget *widget,
 static gboolean button_clicked(GtkWidget *widget,
                                GdkEvent *event,
                                gpointer data);
+static void button_size_allocated(GtkWidget *button,
+                                  GtkAllocation *allocation,
+                                  gpointer user_data);
 
 static void window_opened(WnckScreen *screen,
                           WnckWindow *window,
@@ -168,6 +171,19 @@ static gboolean button_clicked(GtkWidget *widget,
         return TRUE;
 }
 
+static void button_size_allocated(GtkWidget *button,
+                                  GtkAllocation *allocation,
+                                  gpointer user_data)
+{
+        /* Set the icon geometry (used for the minimise animation) */
+        gint x, y;
+        GtkWidget *toplevel = gtk_widget_get_toplevel(button);
+        WnckWindow *bwindow = (WnckWindow*)g_object_get_data(G_OBJECT(button), "bwindow");
+        gtk_widget_translate_coordinates(button, toplevel, 0, 0, &x, &y);
+        gdk_window_get_root_coords(gtk_widget_get_window(toplevel), x, y, &x, &y);
+        wnck_window_set_icon_geometry(bwindow, x, y, allocation->width, allocation->height);
+}
+
 static void window_opened(WnckScreen *screen,
                           WnckWindow *window,
                           gpointer userdata)
@@ -236,6 +252,8 @@ static void window_opened(WnckScreen *screen,
                          G_CALLBACK(button_clicked),
                          self);
 
+        g_signal_connect(button, "size-allocate", G_CALLBACK(button_size_allocated), self);
+
         gtk_box_pack_start(GTK_BOX(self), button, FALSE, FALSE, 0);
         gtk_widget_show_all(button);
 }
-- 
1.9.1

