From 0879b0775647d87096fd1bffa6135ee6bd1487a1 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Wed, 23 Jul 2014 23:58:48 +0100
Subject: [PATCH 8/8] session: Fix incorrect file header

---
 session/BudgieSession.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/session/BudgieSession.vala b/session/BudgieSession.vala
index cc1024e..fc927c4 100644
--- a/session/BudgieSession.vala
+++ b/session/BudgieSession.vala
@@ -1,5 +1,5 @@
 /*
- * Session.vala
+ * BudgieSession.vala
  * 
  * Copyright 2014 Ikey Doherty <ikey.doherty@gmail.com>
  * 
-- 
1.9.1

