From 7ae9cf3d7827ee4aa3b5b78324d603fb8024220a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Thu, 10 Jul 2014 19:13:01 +0100
Subject: [PATCH 11/22] wm: Use faster animations so applications are "ready"
 sooner

---
 wm/plugin.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/wm/plugin.c b/wm/plugin.c
index bdb1083..7b3dd20 100644
--- a/wm/plugin.c
+++ b/wm/plugin.c
@@ -50,11 +50,11 @@
 
 #include "plugin.h"
 
-#define DESTROY_TIMEOUT    200
+#define DESTROY_TIMEOUT    100
 #define DESTROY_SCALE      0.8
-#define MINIMIZE_TIMEOUT   250
-#define MAXIMIZE_TIMEOUT   250
-#define MAP_TIMEOUT        200
+#define MINIMIZE_TIMEOUT   150
+#define MAXIMIZE_TIMEOUT   100
+#define MAP_TIMEOUT        110
 #define MAP_SCALE          0.8
 #define BACKGROUND_TIMEOUT 250
 #define SWITCH_TIMEOUT     500
-- 
1.9.1

