From 7cca81aa57fd36fee21a7383149f244f4421b67b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Thu, 10 Jul 2014 20:39:26 +0100
Subject: [PATCH 14/22] panel: Fix comment

---
 panel/budgie-panel-main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panel/budgie-panel-main.c b/panel/budgie-panel-main.c
index 0ea749d..2c7f5ba 100644
--- a/panel/budgie-panel-main.c
+++ b/panel/budgie-panel-main.c
@@ -53,7 +53,7 @@ gint main(gint argc, gchar **argv)
         app = g_application_new(BUDGIE_PANEL_ID, G_APPLICATION_FLAGS_NONE);
         g_signal_connect(app, "activate", G_CALLBACK(activate), NULL);
 
-        /* Logout action */
+        /* Menu action */
         action = g_simple_action_new(ACTION_MENU, NULL);
         g_signal_connect(action, "activate", G_CALLBACK(menu_cb), app);
         g_action_map_add_action(G_ACTION_MAP(app), G_ACTION(action));
-- 
1.9.1

