From c97ea7441b925f0688163784e75451979f5d93ab Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Fri, 11 Jul 2014 00:55:57 +0100
Subject: [PATCH 18/22] budgie-session-dialog: Rename and re-order items
 logically

---
 session/budgie-session-dialog.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/session/budgie-session-dialog.c b/session/budgie-session-dialog.c
index 9abed12..d2b048e 100644
--- a/session/budgie-session-dialog.c
+++ b/session/budgie-session-dialog.c
@@ -150,31 +150,31 @@ static void budgie_session_dialog_init(BudgieSessionDialog *self)
         gtk_box_pack_start(GTK_BOX(top), label, TRUE, TRUE, 0);
 
         /* Add some buttons to uh.. logout, etc. :) */
-        button = gtk_button_new_with_label("Logout");
+        button = gtk_button_new_with_label("Log Out");
         g_object_set_data(G_OBJECT(button), "action", "logout");
         g_signal_connect(button, "clicked", G_CALLBACK(clicked), self);
         gtk_button_set_relief(GTK_BUTTON(button), GTK_RELIEF_NONE);
         gtk_box_pack_start(GTK_BOX(layout), button, FALSE, FALSE, 0);
 
-        button = gtk_button_new_with_label("Reboot");
-        g_object_set_data(G_OBJECT(button), "action", "reboot");
+        button = gtk_button_new_with_label("Suspend");
+        g_object_set_data(G_OBJECT(button), "action", "suspend");
         g_signal_connect(button, "clicked", G_CALLBACK(clicked), self);
         gtk_button_set_relief(GTK_BUTTON(button), GTK_RELIEF_NONE);
         gtk_box_pack_start(GTK_BOX(layout), button, FALSE, FALSE, 0);
-        if (!can_reboot) {
+        if (!can_suspend) {
                 gtk_widget_set_sensitive(GTK_WIDGET(button), FALSE);
         }
 
-        button = gtk_button_new_with_label("Suspend");
-        g_object_set_data(G_OBJECT(button), "action", "suspend");
+        button = gtk_button_new_with_label("Restart");
+        g_object_set_data(G_OBJECT(button), "action", "reboot");
         g_signal_connect(button, "clicked", G_CALLBACK(clicked), self);
         gtk_button_set_relief(GTK_BUTTON(button), GTK_RELIEF_NONE);
         gtk_box_pack_start(GTK_BOX(layout), button, FALSE, FALSE, 0);
-        if (!can_suspend) {
+        if (!can_reboot) {
                 gtk_widget_set_sensitive(GTK_WIDGET(button), FALSE);
         }
 
-        button = gtk_button_new_with_label("Poweroff");
+        button = gtk_button_new_with_label("Shut Down");
         g_object_set_data(G_OBJECT(button), "action", "poweroff");
         g_signal_connect(button, "clicked", G_CALLBACK(clicked), self);
         gtk_button_set_relief(GTK_BUTTON(button), GTK_RELIEF_NONE);
-- 
1.9.1

