diff -ruNb a/src/common/servlist.c b/src/common/servlist.c
--- a/src/common/servlist.c	2013-04-01 21:07:55.000000000 -0400
+++ b/src/common/servlist.c	2013-06-18 00:33:18.375697589 -0400
@@ -1011,8 +1011,14 @@
 servlist_load_defaults (void)
 {
 	int i = 0, j = 0;
+    
 	ircnet *net = NULL;
-	guint def_hash = g_str_hash ("freenode");
+	guint def_hash = g_str_hash ("SolusOS IRC");
+
+    net = servlist_net_add ("SolusOS IRC", NULL, FALSE);
+    servlist_server_add (net, "irc.solusos.com/6669");
+    net->autojoin = strdup (_("#SolusOS-Offtopic,#SolusOS"));
+    net->flags |= FLAG_AUTO_CONNECT;
 
 	while (1)
 	{
