From 8f4cca933b2f0ffe3b853f9aa59cef56cb224ca0 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solusos.com>
Date: Wed, 11 Sep 2013 14:38:13 +0100
Subject: [PATCH] Apply SolusOS defaults and branding

---
 defaults/defaults |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/defaults/defaults b/defaults/defaults
index 17dc3d7..4734297 100644
--- a/defaults/defaults
+++ b/defaults/defaults
@@ -50,10 +50,10 @@ snap_to_border=true
 snap_to_windows=false
 snap_width=10
 sync_to_vblank=false
-theme=Default
+theme=SolusOS-Darker
 tile_on_move=true
 title_alignment=center
-title_font=Sans Bold 9
+title_font=Droid Sans Bold 9
 title_horizontal_offset=0
 title_shadow_active=false
 title_shadow_inactive=false
-- 
1.7.10.4

